drivers/sensor/maxim/ds3231/ds3231.c

```diff
@@ -73,7 +73,7 @@ static int sensor_ds3231_channel_get(const struct device *dev, enum sensor_chann
 	struct sensor_ds3231_data *data = dev->data;
 
 	switch (chan) {
-	case SENSOR_CHAN_AMBIENT_TEMP:
+	case SENSOR_CHAN_AMBIENT_TEMP: {
 		const uint16_t raw_temp = data->raw_temp;
```

Comment:
Variables need to be declared in the beginning of the function, right under `struct sensor_ds3231_data *data = dev->data;`, otherwise, move out the whole calculation into a helper:
```
static void raw_temp_to_sensor_value()
{
        ...
}

I prefer helpers in any case :)

...

switch (chan) {
case SENSOR_CHAN_AMBIENT_TEMP:
        raw_temp_to_sensor_value();
        break

...
```

Reply:
Mine as well, I'm just surprised that this issue wasn't noticed and reported in the first review. However, I will change this for the 'get' function.

Nevertheless, @gergovari should go through all MFD/RTC/SENS artifacts of the driver again and submit a clean-up PR accordingly, if this is such a hard requirement of the Zephyr Code Guidelines.